File: iOS/AmazonConnectInAppCallingIOSSample/AmazonConnectInAppCallingIOSSample/InAppCallingComponent/Services/ScreenShare/ScreenShare.swift
//
//  ScreenShare.swift
//  AmazonConnectInAppCallingIOSSample
//
//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
//  SPDX-License-Identifier: MIT-0
//

import Foundation
import AmazonChimeSDK

protocol ScreenShare: AnyObject {

    var delegate: ScreenShareDelegate? { get set }

    func startScreenShare()

    func stopScreenShare()

    func addVideoSink(_ videoSink: VideoSink)

    func removeVideoSink(_ videoSink: VideoSink)
}

File: Web/amazon-connect-ccp-web-calling-example/src/components/MyCustomCCP/ConferenceView/DisplayPanel/VideoTileGrid/ScreenShare/ScreenShare.js
//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
//  SPDX-License-Identifier: MIT-0

import contactManager from '../../../../../../services/ContactManager';
import eventBus from '../../../../../../utils/EventBus';

class ScreenShare extends HTMLElement {
  constructor() {
    super();
  }

  get screenShare() {
    return this.querySelector('#screen-share');
  }

  connectedCallback() {
    this.render();
    contactManager.subscribeToScreenShare(this.screenShare);
    eventBus.on('ScreenSharingStateChanged', () => {
      this.updateStyles();
    });
  }

  updateStyles() {
    this.style.display = contactManager.isScreenSharingSessionStarted ? 'inline-block' : 'none';
  }

  render() {
    this.innerHTML = `
      <style>
        #screen-share {
          height: 300px;
          width: 400px;
          background-color: grey;
          position: absolute;
          top: 100px;
          box-sizing: border-box;
        }
      </style>

      <video id="screen-share"></video>
    `;
    this.updateStyles();
  }
}

customElements.define('screen-share', ScreenShare);
